apps/framework-cli/src/cli/watcher.rs

```diff
@@ -340,28 +339,41 @@ async fn watch(
                                 &mut clickhouse_client_v2,
                                 &project,
                             )
-                            .await?;
-
-                            log::info!("Plan Changes: {:?}", plan_result.changes);
-
-                            display::show_changes(&plan_result);
-                            framework::core::execute::execute_online_change(
-                                &project,
-                                &plan_result,
-                                route_update_channel.clone(),
-                                syncing_process_registry,
-                                project_registries,
-                            )
-                            .await?;
-
-                            store_infrastructure_map(
-                                &mut clickhouse_client_v2,
-                                &project.clickhouse_config,
-                                &plan_result.target_infra_map,
-                            )
-                            .await?;
-
-                            Ok(())
+                            .await;
+
+                            match plan_result {
+                                Ok(plan_result) => {
+                                    log::info!("Plan Changes: {:?}", plan_result.changes);
+
+                                    display::show_changes(&plan_result);
+                                    framework::core::execute::execute_online_change(
+                                        &project,
+                                        &plan_result,
+                                        route_update_channel.clone(),
+                                        syncing_process_registry,
+                                        project_registries,
+                                    )
+                                    .await?;
+
+                                    store_infrastructure_map(
+                                        &mut clickhouse_client_v2,
+                                        &project.clickhouse_config,
+                                        &plan_result.target_infra_map,
+                                    )
+                                    .await?;
+
+                                    Ok(())
+                                }
+                                Err(e) => {
+                                    show_message!(MessageType::Error, {
+                                        Message {
+                                            action: "\nFailed".to_string(),
+                                            details: format!("\n {}", e),
+                                        }
+                                    });
+                                    Ok(())
```

Comment:
Both branches return `Ok(())`
what about moving that outside the pattern matching?

Reply:
yep - good catch